Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/FileTransporter.cs

```diff
@@ -73,39 +73,73 @@ public FileTransferResult Run(CancellationToken cancellationToken)
                 client.ConnectionInfo.KeyExchangeAlgorithms.Remove("curve25519-sha256");
                 client.ConnectionInfo.KeyExchangeAlgorithms.Remove("curve25519-sha256@libssh.org");
 
+                var expectedServerFingerprint = _batchContext.Connection.ServerFingerPrint;
                 // Check the fingerprint of the server if given.
-                if (!String.IsNullOrEmpty(_batchContext.Connection.ServerFingerPrint))
+                if (!string.IsNullOrEmpty(expectedServerFingerprint))
                 {
+                    SetCurrentState(TransferState.Connection, "Checking server fingerprint.");
                     try
                     {
-                        SetCurrentState(TransferState.Connection, "Checking server fingerprint.");
-                        // If this check fails then SSH.NET will throw an SshConnectionException - with a message of "Key exchange negotiation failed".
+                        #region HostKeyAuthentication
                         client.HostKeyReceived += delegate (object sender, HostKeyEventArgs e)
                         {
-                            // First try with SHA256 typed fingerprint
-                            using (SHA256 mySHA256 = SHA256.Create())
+                            if (Util.IsMD5(expectedServerFingerprint.Replace(":", "").Replace("-", "")))
                             {
-                                var sha256Fingerprint = Convert.ToBase64String(mySHA256.ComputeHash(e.HostKey));
-                                // Set the userResultMessage in case checking server fingerprint failes.
                                 userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
-                                                        $"Expected fingerprint: '{_batchContext.Connection.ServerFingerPrint}', but was: '{sha256Fingerprint}'.";
-                                e.CanTrust = (sha256Fingerprint == _batchContext.Connection.ServerFingerPrint);
+                                            $"Expected fingerprint: '{expectedServerFingerprint}', but was: '{BitConverter.ToString(e.FingerPrint).Replace("-", "").Replace(":", "")}'.";
+                                e.CanTrust = expectedServerFingerprint == BitConverter.ToString(e.FingerPrint).Replace("-", "").Replace(":", "");
+
+                                if (!e.CanTrust)
+                                {
+                                    userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
+                                                $"Expected fingerprint: '{expectedServerFingerprint}', but was: '{BitConverter.ToString(e.FingerPrint).Replace('-', ':')}'.";
```

Comment:
You can extract the `BitConverter.ToString(e.FingerPrint).Replace('-', ':')` with a variable. This will both help reading the code and also will make sure it is easier to modify later.